frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GraphTableDialog.kt

```diff
@@ -0,0 +1,107 @@
+package com.virtuslab.gitmachete.frontend.actions.dialogs
+
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.ScrollPaneFactory
+import com.intellij.ui.table.JBTable
+import com.intellij.util.ui.JBDimension
+import com.intellij.util.ui.JBUI
+import com.virtuslab.binding.RuntimeBinding
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositorySnapshot
+import com.virtuslab.gitmachete.frontend.ui.api.table.ISimpleGraphTableProvider
+import java.awt.event.ActionEvent
+import java.util.function.Consumer
+import javax.swing.AbstractAction
+
+class GraphTableDialog
+    private constructor(
+        private val table: JBTable,
+        private val repositorySnapshot: IGitMacheteRepositorySnapshot?,
+        private val dimension: JBDimension,
+        private val okAction: Consumer<IGitMacheteRepositorySnapshot>?,
+        private val editAction: Runnable?,
+        private val cancelButtonVisible: Boolean,
+        private val windowTitle: String,
+        private val okButtonText: String
+    ) : DialogWrapper(/* canBeParent */ false) {
+
+  init {
+    super.init()
+    title = windowTitle
+    setOKButtonText(okButtonText)
+  }
+
+  companion object {
+    // @JvmOverloads annotation instructs the Kotlin compiler to generate overloads
+    // for this function that substitute default parameter values (dimension in our case).
+    @JvmOverloads
+    fun of(
+        gitMacheteRepositorySnapshot: IGitMacheteRepositorySnapshot,
+        windowTitle: String,
+        dimension: JBDimension = JBDimension(/* width */ 800, /* height */ 500),
+        emptyTableText: String?,
+        okAction: Consumer<IGitMacheteRepositorySnapshot>?,
+        editAction: Runnable?,
```

Comment:
There's (?) a silent assumption that `okAction` invokes `editAction`... :thinking:

Reply:
y? I do not get your doubts :thinking:

Reply:
that's right. We could build both build "actions" within the `DiscoverAction` and gain the isolation here. Is it your point?

Reply:
Yes, would be nice